subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Checkstyle.java

```diff
@@ -88,6 +87,7 @@ public Checkstyle() {
         // Set default JavaLauncher to current JVM in case
         // CheckstylePlugin that sets Java launcher convention is not applied
         this.javaLauncher = configureFromCurrentJvmLauncher(getToolchainService(), getObjectFactory());
+        getMaxErrors().convention(0);
```

Comment:
🤔 Wonder how we could make this a default convention. Maybe we could introduce an `IntProperty`?